lib/galaxy/authnz/psa_authnz.py

```diff
@@ -118,7 +118,10 @@ def __init__(self, provider, oidc_config, oidc_backend_config, app_config: Galax
 
         self.config[setting_name("USER_MODEL")] = "models.User"
         # Use a custom auth pipeline if configured.
-        auth_pipeline = app_config.get("oidc_auth_pipeline", AUTH_PIPELINE)
+        auth_pipeline = app_config.oidc_auth_pipeline or AUTH_PIPELINE
+        # Add extra steps to the auth pipeline if configured.
+        if app_config.oidc_auth_pipeline_extra:
+            auth_pipeline = auth_pipeline + tuple(app_config.oidc_auth_pipeline_extra)
```

Comment:
@marius-mather A related issue - what about the refresh pipeline? How do we have a similarly pluggable method for handling refreshes? I'm thinking in particular about the need for syncing roles whenever the access token is refreshed.

Reply:
Good point - I'll have to have a think about this. Maybe best handled in a separate PR though